lib/ZumoOta/src/NetworkCustom.cpp

```diff
@@ -0,0 +1,125 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+/*******************************************************************************
+    DESCRIPTION
+*******************************************************************************/
+/**
+ * @brief  ZumoOta application
+ * @author Decareme Pauline Ngangnou <ngandeca@yahoo.fr>
+ */
+
+/******************************************************************************
+ * Includes
+ *****************************************************************************/
+#include "NetworkCustom.h"
+#include <WiFi.h>
+#include <Logging.h>
+#include "MySettings.h"
+
+/******************************************************************************
+ * Compiler Switches
+ *****************************************************************************/
+
+/******************************************************************************
+ * Macros
+ *****************************************************************************/
+
+/******************************************************************************
+ * Types and classes
+ *****************************************************************************/
+
+/******************************************************************************
+ * Prototypes
+ *****************************************************************************/
+
+/******************************************************************************
+ * Local Variables
+ *****************************************************************************/
+MySettings settings;
+/******************************************************************************
+ * Public Methods
+ *****************************************************************************/
+NetworkCustom::NetworkCustom()
+{
+    m_connected = false;
+    m_connectAttempts = 0;
+}
+
+NetworkCustom::~NetworkCustom()
+{
+    
+}
+
+void NetworkCustom::connectToWiFi()
+{
+    WiFi.mode(WIFI_STA);
+    WiFi.begin(settings.getWiFiSSID(), settings.getWiFiPassword());
+    delay(10000);

```

Comment:
why ??

Reply:
I need to implement a delay to check if the WiFi status is connected. After three consecutive attempts, it should configure to a local WiFi.

Reply:
This is already implemented in Network. Why not using it instead?